### ISSUE TITLE
Informal spellings in specifications

### ISSUE THREAD

**Issue by SebastianZ** [w3c/csswg-drafts#5850]
For some time now informal spellings for commonly used words made it into the specifications. Some examples are "thru", "thruout", and "tho". Examples for specifications using them are [CSS Environment Variables 1](https://drafts.csswg.org/css-env-1/#intro), [CSS Syntax 3](https://drafts.csswg.org/css-syntax-3/#tokenizer-definitions), and [CSS Images 4](https://drafts.csswg.org/css-images-4/#image-set-notation).

As a non-native English speaker who learned English at school, it initially took me a moment to understand what those words actually mean. And my personal thought is that technical specifications, and especially those read by a global audience, should strictly keep to a formal language. As I know that @tabatkins is an advocate of this informal spelling, I already wrote him privately about this some time ago.

His answer back then was:

> "Thru" (and "tho") is a perfectly cromulent spelling. it's considered
informal at this point, but it's been strongly advocated by spelling
reformers in the past, and is specifically used in some technical
journal-ese. It's not prone to non-native misunderstanding; if you
look up "thru" or "tho" it immediately comes up as a spelling variant
of "through"/"though". At worst people might misread it as a spelling
variant of "threw", but that's nonsensical in context and thus not
worth worrying about.
> 
> While informal, it's definitely "correct" English, just a spelling
you're not used to. But I feel strongly that the "silent gh" needs to
die in English, and no longer spell words with it when I can help it.

Though today I saw @dholbert's [patch for fixing "typos" in CSS Images 4](https://github.com/w3c/csswg-drafts/pull/5848) changing "tho" to "though".

Therefore I thought (or as Tab would probably spell it, "thot" 😉), I'd bring this up to a broader audience.

Should it be allowed or even be encouraged to use colloquial or informal spellings and words in specifications?

Sebastian

**Comment by plinss** [comment]
FWIW the TAG has a writing style guide: https://github.com/w3ctag/process/blob/master/style-guide.md *

It's (currently) silent on the subject of neologisms (however cromulent they may be), but keeping in the spirit, the question should be "does this word make the document easier to understand or harder?" I accept the informal spellings make the document easier to *write*, but that's not what we should be optimizing for.

I'd like to hear from more non-native English speakers on this topic.

Also, pinging @alice for her input (the author of the TAG style guide).

* Note that at this point the guide is directed towards the TAG, and is not meant to be an authoritative style guide for all of W3C, it might be a good foundation for one tho...

**Comment by rachelandrew** [comment]
With my "editor of technical material written for a global audience" hat on, I'd suggest to avoid these spellings. They are unlikely to be the English people have learned, and specs are hard enough to understand as it is.

I am not a fan of the idea that people could search to find out what they meant. If there is a word that people are likely to understand and not need to search for the meaning of, it makes more sense to use that word. Formal usage is more likely to be globally understood. Things do change, it used to be that contractions were said to be bad practice, but - as in the TAG guide - more style guides encourage the most common contractions. So if there is actual data on these words being better, that's a discussion to have. I think it would be good if we are consistent however.

As far as style guides for technical content go, the Microsoft Style Guide is very good https://docs.microsoft.com/en-us/style-guide/welcome/ but I don't think I have seen anything in there that addresses this particular issue.

**Comment by alice** [comment]
I can only second everything @rachelandrew said, and agree with @plinss that we should be aiming for readability. 

The advice on contractions is based on advice from professional copy-editors that they improve readability, e.g. https://stroppyeditor.wordpress.com/2015/10/12/contractions-which-are-common-and-which-arent/ (which is also linked in the style guide), and backed up by anecdotal reports from speakers of English as a second language.

It would be good to find analogous evidence for or against non-standard spelling; the fact that @SebastianZ filed this issue is a strong start in the "against" direction.

My personal opinion is that a technical spec isn't the place to try and move the needle on not-yet-standardard spellings like those mentioned above: specs are tough enough to read already, without added speed bumps from non-standard spelling.

As @plinss noted, we don't have any advice for spec language currently, and creating a style guide for specs would be a big project. However, honestly I think we probably only need to convince one or possibly two people to change practices in this specific instance.

**Comment by frivoal** [comment]
I'm not opposed to linguistic activism in general, but I think specifications are a poor place to do it. As a non native speaker of English, who is nonetheless reasonably fluent, I'd like to mention the effect this sort of things has had on me: broadly speaking I have not faced particular difficulties understanding documents that used informal language (including unusual spellings, colloquialisms, in-jokes, jargon, etc). However, since reading W3C specifications represent a significant percentage of my exposure to English, in the absence of contradicting information, I tend to assume that whatever is common in such documents is "normal written English". Occasionally though, something I learned there is not (yet?) accepted as mainstream or formal English, and when I later reuse these words in other contexts, it can lead to people being surprised or even confused at the language I use.

To some degree, this achieves Tab's goal: if people start to use these spellings after seeing them, they spread, which is what he wants. But spreading them through people for who English is a second (or third, or…) language isn't at no cost to those people if it occasionally causes additional communication difficulties or social faux pas.

**Comment by svgeesus** [comment]
> But I feel strongly that the "silent gh" needs to
die in English, and no longer spell words with it when I can help it.

CSS specifications are an unsuitable venue to promote _possible future spelling reforms_.

> since reading W3C specifications represent a significant percentage of my exposure to English, in the absence of contradicting information, I tend to assume that whatever is common in such documents is "normal written English"

Yes, that is the normalizing, language activism part.

> a technical spec isn't the place to try and move the needle on not-yet-standardard spellings

I agree. If the intent is to build up a body of published work which can then be pointed to as evidence of usage, this can be easily accomplished by personal blogs and the like.

**Comment by fantasai** [comment]
Thanks for raising the issue @SebastianZ :) I think we had a good discussion.

**Comment by SebastianZ** [comment]
Thank you all for clarifying this! I guess there's nothing left to say, so I close this issue now.

Sebastian
